phpunit/class-wp-rest-term-search-handler-test.php

```diff
@@ -0,0 +1,248 @@
+		$term_ids = array(
+			self::$my_category,
+			self::$my_tag,
+		);
+
+		foreach ( $term_ids as $term_id ) {
+			wp_delete_term( $term_id, true );
+		}
+	}
+
+
+	/**
+	 * Search through terms of any type.
+	 */
+	public function test_get_items_search_type_term() {
+		$response = $this->do_request_with_params(
+			array(
+				'per_page' => 100,
+				'type'     => 'term',
+			)
+		);
+		$this->assertEquals( 200, $response->get_status() );
+		$this->assertEqualSets(
+			array(
+				0 => 1, // That is the default category.
+				self::$my_category,
+				self::$my_tag,
+			),
+			wp_list_pluck( $response->get_data(), 'id' )
+		);
+	}
+
+	/**
+	 * Search through terms of subtype 'category'.
+	 */
+	public function test_get_items_search_type_term_subtype_category() {
+		$response = $this->do_request_with_params(
+			array(
+				'per_page' => 100,
+				'type'     => 'term',
+				'subtype'  => 'category',
+			)
+		);
+
+		$this->assertEquals( 200, $response->get_status() );
+		$this->assertEqualSets(
+			array(
+				0 => 1, // That is the default category.
+				self::$my_category,
+			),
+			wp_list_pluck( $response->get_data(), 'id' )
+		);
+	}
+
+	/**
+	 * Search through posts of an invalid post type.
+	 */
+	public function test_get_items_search_term_subtype_invalid() {
+		$response = $this->do_request_with_params(
+			array(
+				'per_page' => 100,
+				'type'     => 'term',
+				'subtype'  => 'invalid',
+			)
+		);
+
+		$this->assertErrorResponse( 'rest_invalid_param', $response, 400 );
+	}
+
+	/**
+	 * Search through posts and pages.
+	 */
+	public function test_get_items_search_categories_and_tags() {
+		$response = $this->do_request_with_params(
+			array(
+				'per_page' => 100,
+				'type'     => 'term',
+				'subtype'  => 'category,post_tag',
+			)
+		);
+		$this->assertEquals( 200, $response->get_status() );
+		$this->assertEqualSets(
+			array(
+				0 => 1, // This is the default category.
+				self::$my_category,
+				self::$my_tag,
+			),
+			wp_list_pluck( $response->get_data(), 'id' )
+		);
+	}
+
+	/**
+	 * Search through all that matches a 'Test Category' search.
+	 */
+	public function test_get_items_search_for_testcategory() {
+		$response = $this->do_request_with_params(
+			array(
+				'per_page' => 100,
+				'search'   => 'Test Category',
+				'type'     => 'term',
+			)
+		);
+
+		$this->assertEquals( 200, $response->get_status() );
+		$this->assertEqualSets(
+			array(
+				self::$my_category,
+			),
+			wp_list_pluck( $response->get_data(), 'id' )
+		);
+	}
+
+	/**
+	 * Search through all that matches a 'Test Tag' search.
+	 */
+	public function test_get_items_search_for_testtag() {
+		$response = $this->do_request_with_params(
+			array(
+				'per_page' => 100,
+				'search'   => 'Test Tag',
+				'type'     => 'term',
+			)
+		);
+
+		$this->assertEquals( 200, $response->get_status() );
+		$this->assertEqualSets(
+			array(
+				self::$my_tag,
+			),
+			wp_list_pluck( $response->get_data(), 'id' )
+		);
+	}
+
```

Comment:
Can we have a test where the search doesn't match?

Reply:
Yes we can! 7408a1b861